### PR TITLE
Added verification of tasks/thinexecutions in DataflowOAuthIT.

### PR DESCRIPTION
**Pull request by corneil**

### CODE REVIEW

File: spring-cloud-dataflow-server/src/test/java/org/springframework/cloud/dataflow/integration/test/oauth/DataflowOAuthIT.java [hunk 1]
```diff
@@ -54,12 +56,34 @@ public void testSecuredSetup() throws Exception {
...
+									VersionUtils.getThreePartVersion(version), "2.11.3")) {
+								api = "tasks/thinexecutions";
+							}
+							log.info("Checking {}", api);
+							cmdResult = execInToolsContainer("curl", "-u", "janne:janne", "http://dataflow:9393/" + api);
+							response = cmdResult.getStdout();
+							log.debug("Response is {}", response);
+							ok = !JsonPath.parse(response).read("$._links.self.href", String.class).isEmpty();
+
+							// TODO add checks for new endpoints to check security
```
Comment (onobc) [inline spring-cloud-dataflow-server/src/test/java/org/springframework/cloud/dataflow/integration/test/oauth/DataflowOAuthIT.java:80]: This does add coverage for the `thin/executions` endpoint but what is the expectation for newly added endpoints? How will we know to abide by this comment? 

The issue that we ran into was that PRO was out of sync w/ OSS endpoint mappings. This does not fill that gap.

**Review by onobc — CHANGES_REQUESTED** [review]
Thx @corneil - a few comments to address.

**Review by cppwfs — COMMENTED** [review]
I agree with Chris' requests.  No further additions from my side.

File: spring-cloud-dataflow-server/src/test/java/org/springframework/cloud/dataflow/integration/test/oauth/DataflowOAuthIT.java (hunk 1 above)
Comment (corneil) [reply spring-cloud-dataflow-server/src/test/java/org/springframework/cloud/dataflow/integration/test/oauth/DataflowOAuthIT.java:80]: This test is executed as part of ci-it-security.
The Pro will be covered by the endpoint test running in CF ATs.

File: spring-cloud-dataflow-server/src/test/java/org/springframework/cloud/dataflow/integration/test/oauth/DataflowOAuthIT.java [hunk 2]
```diff
@@ -54,12 +56,34 @@ public void testSecuredSetup() throws Exception {
...
+						}
+						catch (Throwable x) {
+							log.error("Exception:" + x, x);
```
[the remark this answers is not in the document]
Comment (corneil) [reply spring-cloud-dataflow-server/src/test/java/org/springframework/cloud/dataflow/integration/test/oauth/DataflowOAuthIT.java]: Remove log and try / catch

File: spring-cloud-dataflow-server/src/test/java/org/springframework/cloud/dataflow/integration/test/oauth/DataflowOAuthIT.java (hunk 1 above)
Comment (onobc) [reply spring-cloud-dataflow-server/src/test/java/org/springframework/cloud/dataflow/integration/test/oauth/DataflowOAuthIT.java]: > The Pro will be covered by the endpoint test running in CF ATs.

Which endpoint test? The point of https://github.com/pivotal/scdf-pro/issues/192 is to find out issues prior to CF ATs (in the PRO tests).

File: spring-cloud-dataflow-server/src/test/java/org/springframework/cloud/dataflow/integration/test/oauth/DataflowOAuthIT.java [hunk 3]
```diff
@@ -63,26 +63,19 @@ public void testSecuredSetup() throws Exception {
...
+						if (VersionUtils.isDataFlowServerVersionGreaterThanOrEqualToRequiredVersion(
+								VersionUtils.getThreePartVersion(version), "2.11.3")) {
+							api = "tasks/thinexecutions";
 						}
+						log.info("Checking {}", api);
+						cmdResult = execInToolsContainer("curl", "-u", "janne:janne", "http://dataflow:9393/" + api);
+						response = cmdResult.getStdout();
+						log.debug("Response is {}", response);
+						ok = !JsonPath.parse(response).read("$._links.self.href", String.class).isEmpty();
+						// TODO add checks for new endpoints to check security
```
Comment (onobc) [inline spring-cloud-dataflow-server/src/test/java/org/springframework/cloud/dataflow/integration/test/oauth/DataflowOAuthIT.java:78]: I don't think this test is the place to add endpoints to. I still would like to remove this comment.

File: spring-cloud-dataflow-server/src/test/java/org/springframework/cloud/dataflow/integration/test/oauth/DataflowOAuthIT.java [hunk 4]
```diff
@@ -54,12 +56,28 @@ public void testSecuredSetup() throws Exception {
...
+					String username = JsonPath.parse(response).read("$.securityInfo.username", String.class);
+					boolean ok = Boolean.TRUE.equals(authenticated) && "janne".equals(username);
 					log.info("Check for oauth {}", ok);
+					if (ok) {
+						String version = JsonPath.parse(response).read("$.versionInfo.core.version");
+						log.info("Version=[{}]", version);
+						String api = "tasks/executions";
+						if (VersionUtils.isDataFlowServerVersionGreaterThanOrEqualToRequiredVersion(
+								VersionUtils.getThreePartVersion(version), "2.11.3")) {
+							api = "tasks/thinexecutions";
```
Comment (cppwfs) [inline spring-cloud-dataflow-server/src/test/java/org/springframework/cloud/dataflow/integration/test/oauth/DataflowOAuthIT.java:73]: I'm not sure why we are testing this endpoint in the DataflowOauthIT test.    It seems like it should be in a different test.   WDYT?

Comment (corneil) [reply spring-cloud-dataflow-server/src/test/java/org/springframework/cloud/dataflow/integration/test/oauth/DataflowOAuthIT.java:73]: Checking some basic security on endpoints. As we add endpoints we can add basic checks to ensure the endpoint security is properly configured.

File: spring-cloud-dataflow-server/src/test/java/org/springframework/cloud/dataflow/integration/test/oauth/DataflowOAuthIT.java (hunk 1 above)
Comment (corneil) [reply spring-cloud-dataflow-server/src/test/java/org/springframework/cloud/dataflow/integration/test/oauth/DataflowOAuthIT.java]: I added ci-it-security to scdf-pro to perform similar checks on added endpoints to ensure they are protected.

**Comment by corneil** [comment]
Recreating from personal repo

**Pull request closed without merging**